numcodecs/gzip.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, division
+import zlib as _zlib
+
+
+import numpy as np
+
+
+from .abc import Codec
+from .compat import buffer_copy, handle_datetime, buffer_tobytes, PY2
+
+
+class GZip(Codec):
+    """Codec providing gzip compression using zlib via the Python standard library.
+
+    Parameters
+    ----------
+    level : int
+        Compression level.
+
+    """
+
+    codec_id = 'gzip'
+
+    def __init__(self, level=1):
+        self.level = level
+
+    def encode(self, buf):
+
+        # deal with lack of buffer support for datetime64 and timedelta64
+        buf = handle_datetime(buf)
+
+        if isinstance(buf, np.ndarray):
+
+            # cannot compress object array
+            if buf.dtype == object:
+                raise ValueError('cannot encode object array')
+
+            # if numpy array, can only handle C contiguous directly
+            if not buf.flags.c_contiguous:
+                buf = buf.tobytes(order='A')
+
+        if PY2:  # pragma: py3 no cover
+            # ensure bytes, PY2 cannot handle things like bytearray
+            buf = buffer_tobytes(buf)
+
+        # do compression
+        compress = _zlib.compressobj(self.level, wbits=16 + 15)
```

Comment:
Looks like Python 2 doesn't support keyword arguments. Fortunately it does support `wbits`, it just comes as the third positional argument. Something like `_zlib.compressobj(self.level, _zlib.DEFLATED, 16 + 15)` should do the trick and work on both Python 2 and Python 3.